### ISSUE TITLE
"module": "node16" error: `This expression is not callable`

### ISSUE THREAD

**Issue by karlhorky** [postcss/postcss#1814]
Hi there 👋 First of all, thanks for your continued effort on PostCSS and the surrounding community, really amazing!

Using the `"module": "node16"` configuration option in `tsconfig.json` within an ES Module causes the following TypeScript error:

```bash
$ yarn tsc
index.ts:3:1 - error TS2349: This expression is not callable.
  Type 'typeof import("/home/projects/node-vqx4zh/node_modules/postcss/lib/postcss")' has no call signatures.

3 postcss([]);
  ~~~~~~~


Found 1 error in index.ts:3

error Command failed with exit code 2.
```

Demo on StackBlitz - run `yarn tsc` in the Terminal at the bottom to reproduce:

https://stackblitz.com/edit/node-vqx4zh?file=tsconfig.json,package.json,index.ts

![Screenshot 2023-01-29 at 18 30 24](https://user-images.githubusercontent.com/1935696/215345085-a15fef38-95ac-43c5-a25d-57cd92135d41.png)

## Proposed Solution

One way that seems to work to fix the problem is to use `export = postcss` instead of `export default postcss` here:

https://github.com/postcss/postcss/blob/8834314ff334ea8321bf2c83934f276d379512cd/lib/postcss.d.ts#L473

But maybe there is a more elegant way of doing this...

cc @andrewbranch in case I still have the proposed solution wrong over here

## Background

See more information about the problem of misconfigured declaration files in packages here:

- https://github.com/microsoft/TypeScript/issues/49160#issuecomment-1137482639
- https://github.com/microsoft/TypeScript/issues/49189#issuecomment-1137756847

**Comment by ai** [comment]
Will `export = postcss` work in ESM project?

**Comment by karlhorky** [comment]
Yeah the example above is an ESM project

**Comment by andrewbranch** [comment]
The correct solution is to provide an `index.d.mts` file. Take a look at the problem reported by https://arethetypeswrong.github.io. You can read a bit more about that class of problem at https://github.com/arethetypeswrong/arethetypeswrong.github.io. The tool is still in super early days; I hope it will eventually be able to provide a bit more context and suggest a solution. Let me know if it’s at all helpful though!

**Comment by karlhorky** [comment]
Here are the errors from https://arethetypeswrong.github.io/:

`postcss` + `node16 (from ESM)`: 

```
🎭 Imports of "postcss" under the node16 module resolution setting when the importing module is ESM (its extension is .mts or .mjs, or it has a .ts or .js extension and is in scope of a package.json that contains "type": "module") resolved to CJS types, but ESM implementations.
```

<img width="826" alt="Screenshot 2023-02-01 at 12 47 09" src="https://user-images.githubusercontent.com/1935696/216035185-6ea753d9-6ed1-419e-acc6-2b399aae5100.png">

--- 

`postcss/lib/map-generator`, `postcss/lib/parser`, `postcss/lib/symbols`, `postcss/lib/terminal-highlight`, `postcss/lib/tokenize`, `postcss/lib/warn-once`

```
❌ Imports of multiple entrypoints under all module resolution settings resolved to JavaScript files, but no types.
```

<img width="1120" alt="Screenshot 2023-02-01 at 12 49 00" src="https://user-images.githubusercontent.com/1935696/216035553-96b8c39a-729c-41f5-93eb-639e19388272.png">

<img width="805" alt="Screenshot 2023-02-01 at 12 49 11" src="https://user-images.githubusercontent.com/1935696/216035559-18128d5e-9891-46ce-b37d-b668ebb2f8af.png">

**Comment by ai** [comment]
Please, send PR with the simplest way to fix the support

**Comment by remcohaszing** [comment]
I’m working on this. It’s not hard, but I have to make quite a lot of small tweaks manually. PR coming tomorrow!

**Comment by ai** [comment]
The fix by @remcohaszing was released in 8.4.22.

**Comment by karlhorky** [comment]
Nice, lots of green checkmarks on [Are The Types Wrong?](https://arethetypeswrong.github.io/?p=postcss%408.4.22) 🎉 great job @remcohaszing and thanks for the review, merge and publish @ai !
